paperqa/llms.py

```diff
@@ -824,16 +887,65 @@ async def similarity_search(
 
 
 def embedding_model_factory(embedding: str, **kwargs) -> EmbeddingModel:
-    if embedding.startswith("hybrid"):
-        embedding_model_name = "-".join(embedding.split("-")[1:])
-        return HybridEmbeddingModel(
-            models=[
-                LiteLLMEmbeddingModel(name=embedding_model_name),
-                SparseEmbeddingModel(**kwargs),
-            ]
+    """
+    Factory function to create an appropriate EmbeddingModel based on the embedding string.
+
+    Supports:
+    - SentenceTransformer models prefixed with "st-" (e.g., "st-multi-qa-MiniLM-L6-cos-v1")
+    - LiteLLM models (default if no prefix is provided)
+    - Hybrid embeddings prefixed with "hybrid-", contains a sparse and a dense model
+
+    Args:
+        embedding: The embedding model identifier. Supports prefixes like "st-" for SentenceTransformer
+                         and "hybrid-" for combining multiple embedding models.
+        **kwargs: Additional keyword arguments for the embedding model.
+
+    Returns:
+        EmbeddingModel: An instance of a subclass of EmbeddingModel.
```

Comment:
```suggestion
       An instance of a subclass of EmbeddingModel.
```

Don't restate types in docstring if it's in the signature